main.go

```diff
@@ -64,68 +50,74 @@ func applyVirtualMachineExport(vmNamespace, vmName string) error {
 		},
 	}
 
-	_, err = client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
+	_, err := client.VirtualMachineExport(vmNamespace).Create(context.Background(), vmExport, metav1.CreateOptions{})
 	return err
 }
 
-func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
-	log.Printf("Downloading disk image from %s Virtual Machine...\n", vmName)
-
-	cmd := exec.Command("usr/bin/virtctl", "vmexport", "download", vmName, "--vm", vmName, "--volume", volumeName, "--output", diskPath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+func getRawDiskUrlFromVirtualMachineExport(client kubecli.KubevirtClient, vmNamespace, vmName, volumeName string) (string, error) {
+	log.Println("Waiting for VirtualMachineExport to be ready...")
 
-	if err := cmd.Run(); err != nil {
-		return err
+	vmExport, err := getVirtualMachineExportOnceReady(client, vmNamespace, vmName)
+	if err != nil {
+		return "", err
 	}
 
-	if fileInfo, err := os.Stat(diskPath); err != nil || fileInfo.Size() == 0 {
-		return fmt.Errorf("File does not exist or is empty.")
+	if vmExport.Status.Links == nil {
```

Comment:
just a nit, you can simplify this to:
`if vmExport.Status.Links == nil && vmExport.Status.Links.Internal == nil{`